security/pkg/platform/onprem.go

```diff
@@ -84,6 +89,11 @@ func (ci *OnPremClientImpl) GetServiceIdentity() (string, error) {
 		return "", err
 	}
 	if len(serviceIDs) != 1 {
+		for _, s := range serviceIDs {
+			if strings.HasPrefix(s, "spiffe://") {
+				return serviceIDs[0], nil
+			}
+		}
 		return "", fmt.Errorf("cert has %v SAN fields, should be 1", len(serviceIDs))
```

Comment:
Change the error message? I.e., "Cert does not have SPIFFE identity in SAN"

Reply:
done